pkg/ck8s/workload_cluster.go

```diff
@@ -43,8 +43,10 @@ type WorkloadCluster interface {
 	ClusterStatus(ctx context.Context) (ClusterStatus, error)
 	UpdateAgentConditions(ctx context.Context, controlPlane *ControlPlane)
 	UpdateEtcdConditions(ctx context.Context, controlPlane *ControlPlane)
-	NewControlPlaneJoinToken(ctx context.Context, authToken string, microclusterPort int, name string) (string, error)
-	NewWorkerJoinToken(ctx context.Context, authToken string, microclusterPort int, name string) (string, error)
+	NewControlPlaneJoinToken(ctx context.Context, name string) (string, error)
+	NewWorkerJoinToken(ctx context.Context, name string) (string, error)
```

Comment:
final note, let's drop the `name` from the interface here. We expect this to always be `""` so let's hardcode it into the implementation. then callers don't need to care.

_If we ever change this_, we'll add a V2, or change the interface, then callers must adapt and we don't miss any required updates in the code.